ToolChain/Assembler/Parser/Parser.py

```diff
@@ -162,239 +179,128 @@ def _parseCodeLine(self, line: list=None):
             # We keep the comment indicator so we can later discard the instruction.
-            # Put text into instructionCode since this is an impossible case... Easy to
-            # follow in calling code. And put in some indicator for the linker.
-            buildInstruction.instructionCode = line[0].replace(MEMORY_REFERENCE_INDICATORS, "").upper()
-            return
+            # We put the text into instructionCode since this is an impossible case,
+            # and put in some indicator for the linker.
+            buildInstruction.instructionCode = sline[0].replace(MEMORY_REFERENCE_INDICATORS, "").upper()
 
-        # First thing is either a mnemonic or a data identifier or a global memory reference
-        buildInstruction.operationMnemonic = line[0].upper()
-
-        if EXPORTED_REFERENCE_INDICATOR in buildInstruction.operationMnemonic:
-            # This is a memory reference... However, this one is on a global scale (exported)
-            # this is a special case
-            buildInstruction.instructionCode = line[0].upper() + " "
-            buildInstruction.instructionCode += line[1].replace(MEMORY_REFERENCE_INDICATORS, "").upper()
             return
-        if DATA_NUMERIC_INDICATOR in buildInstruction.operationMnemonic:
-            # Treat this as a numeric field!
-            # The numeric value will be but in source immediate value
-            numericValue = self.translateTextImmediateToImmediate(line[1])
-            buildInstruction.sourceImmediate = numericValue
-            buildInstruction.instructionLength = 4
-            buildInstruction.instructionCode = line[0].upper()
-            return
-        if DATA_ALPHA_INDICATOR in buildInstruction.operationMnemonic:
-            # Treat this as an alphabetic field!
-            # The alpha value will be put in the source immediate value
-            # We need to rebuild the alpha into a byte string.
-            alpha = b""
-            for part in line[1:]:
-                alpha += part.encode("utf-8") + b" "
-            alpha = alpha[0:-1]  # Remove trailing space
-            alpha += b"\x00"  # Strings are null terminated
-            buildInstruction.sourceImmediate = alpha
-            buildInstruction.instructionLength = len(alpha)
-            buildInstruction.instructionCode = line[0].upper()
-            return
-        if DATA_MEMORY_REFERENCE in buildInstruction.operationMnemonic:
-            # This is a memory reference used as a constant that needs to be
-            # linked later in the process.
-            if line[1][0] == ":":
-                line[1] = line[1][1:]
-            buildInstruction.sourceImmediate = (":" + line[1] + ":").upper().encode("utf-8")
-            buildInstruction.instructionLength = 4
-            buildInstruction.instructionCode = DATA_MEMORY_REFERENCE
+
+        if buildInstruction.operationMnemonic in indicator_list:
```

Comment:
I mean, some parts of this I like but overall it still uses these indicators to skip ahead rather than left-right parsing.

I'd strongly recommend someone implement an LR0 for this. Would be a good first commit.